modules/svg/src/SkSVGFilter.cpp

```diff
@@ -25,10 +25,18 @@ bool SkSVGFilter::parseAndSetAttribute(const char* name, const char* value) {
                    "primitiveUnits", name, value));
 }
 
+// NON-SKIA-UPSTREAMED CHANGE
+void SkSVGFilter::applyProperties(SkSVGRenderContext* ctx) const { this->onPrepareToRender(ctx); }
+// END OF NON-SKIA-UPSTREAMED CHANGE
+
 sk_sp<SkImageFilter> SkSVGFilter::buildFilterDAG(const SkSVGRenderContext& ctx) const {
     sk_sp<SkImageFilter> filter;
     SkSVGFilterContext fctx(ctx.resolveOBBRect(fX, fY, fWidth, fHeight, fFilterUnits),
                             fPrimitiveUnits);
+    // NON-SKIA-UPSTREAMED CHANGE
+    SkSVGRenderContext lctx(ctx);
```

Comment:
I introduced an additional local `SkSVGRenderContext` here because that's what Skia does for the child nodes before propagating their properties into it. They left a comment explaining that they don't want to modify the original context because it doesn't participate in the normal `onRender()` path, so I decided to play it safe and also avoid modifying the original context while propagating the properties of the `<filter>` element. The naming is not the best, I'd prefer to make this one `localCtx` and rename the child one into `localChildCtx`, but I also wanted to minimise the diff.